src/net.cpp

```diff
@@ -1278,7 +1278,8 @@ void ThreadSocketHandler()
                     }
                     if (fDelete) {
                         vNodesDisconnected.remove(pnode);
-                        assert(std::find(vNodes.begin(),vNodes.end(), pnode) == vNodes.end());  // make sure it has been removed
+                        // no need to remove from vNodes. we know pnode has already been removed from vNodes since that
```

Comment:
true, and this is why I added that assert in the first place.   However, in production code, it seems like overkill to iterate through the list yet again just to prove what happened 10 lines above